tower_cli/resources/node.py

```diff
@@ -0,0 +1,159 @@
+        type=types.Related('inventory'), required=False, display=False)
+    credential = models.Field(
+        type=types.Related('credential'), required=False, display=False)
+    job_type = models.Field(required=False, display=False)
+    job_tags = models.Field(required=False, display=False)
+    skip_tags = models.Field(required=False, display=False)
+    limit = models.Field(required=False, display=False)
+
+    def __getattribute__(self, attr):
+        method = super(Resource, self).__getattribute__(attr)
+        if attr in ['create', 'modify', 'list']:
+            return unified_job_template_options(method)
+        return method
+
+    @staticmethod
+    def _forward_rel_name(rel):
+        return '{0}_nodes'.format(rel)
+
+    @staticmethod
+    def _reverse_rel_name(rel):
+        return 'workflowjobtemplatenodes_{0}'.format(rel)
+
+    def _parent_filter(self, parent, relationship, **kwargs):
+        """
+        Returns filtering parameters to limit a search to the children
+        of a particular node by a particular relationship.
+        """
+        if parent is None or relationship is None:
+            return {}
+        parent_filter_kwargs = {}
+        query_params = ((self._reverse_rel_name(relationship), parent),)
+        parent_filter_kwargs['query'] = query_params
+        if kwargs.get('workflow_job_template', None) is None:
+            parent_data = self.read(pk=parent)['results'][0]
+            parent_filter_kwargs['workflow_job_template'] = parent_data[
+                'workflow_job_template']
+        return parent_filter_kwargs
+
+    @unified_job_template_options
+    def _get_or_create_child(self, parent, relationship, **kwargs):
+        ujt_pk = kwargs.get('unified_job_template', None)
+        if ujt_pk is None:
+            raise exceptions.BadRequest(
+                'A child node must be specified by one of the options '
+                'unified-job-template, job-template, project, or '
+                'inventory-source')
+        kwargs.update(self._parent_filter(parent, relationship, **kwargs))
+        response = self.read(
+            fail_on_no_results=False, fail_on_multiple_results=False, **kwargs)
+        if len(response['results']) == 0:
+            debug.log('Creating new workflow node.', header='details')
+            return client.post(self.endpoint, data=kwargs).json()
+        else:
+            return response['results'][0]
+
+    def _assoc_or_create(self, relationship, parent, child, **kwargs):
+        if child is None:
+            child_data = self._get_or_create_child(
+                parent, relationship, **kwargs)
+            return child_data
+        return self._assoc(self._forward_rel_name(relationship),
+                           parent, child)
+
+    @resources.command
+    @unified_job_template_options
+    @click.argument('parent', type=types.Related('node'))
+    @click.argument('child', type=types.Related('node'), required=False)
```

Comment:
In my opinion, child node should not be omitted in (dis)association (API claims with only one node. Plus, what would be the point just providing one side of a binary relationship?)